docs/Talkbuchet.js

```diff
@@ -0,0 +1,609 @@
+ * saturated before Janus is. The script will write to the console when a peer
+ * could not be connected or if it was disconnected after being connected; if
+ * there is a large number of those messages or the CPU consumption is very high
+ * the client has probably reached its limit.
+ *
+ * Besides the messages written by the script itself you can manually check the
+ * connection state by copying and pasting in the browser console the following
+ * commands:
+ * - For the publishers:
+ * Object.values(publishers).forEach(publisher => { console.log(publisher.peerConnection.iceConnectionState) })
+ * - For the subscribers:
+ * subscribers.forEach(subscriber => { console.log(subscriber.peerConnection.iceConnectionState) })
+ *
+ * DISCLAIMER:
+ * -----------------------------------------------------------------------------
+ * Talk performs some optimizations during calls, like reducing the video
+ * quality based on the number of participants. This script does not take into
+ * account those things; it is meant to be used for stress tests of Janus rather
+ * than to accurately simulate Talk behaviour.
+ *
+ * Independently of that, please note that an accurate simulation would not be
+ * possible given that a single client has to behave as several different
+ * clients. In a real call each client could have a different behaviour (not
+ * only due to sending different media streams, but also from having different
+ * CPU and network conditions), and that might even also affect Janus and the
+ * server regarding very low level things (but relevant for performance on
+ * highly loaded systems) like caches.
+ *
+ * Nevertheless, if those things are kept in mind the script can be anyway used
+ * as a rough performance test of specific call scenarios. Just remember that in
+ * regular calls peer connections increase quadratically with the number of
+ * participants; specifically, publishers increase linearly while subscribers
+ * increase quadratically.
+ *
+ * For example a call between 10 participants has 10 publishers and 90
+ * subscribers (9 for each publisher) for a total of 100 peer connections, while
+ * a call between 30 participants has 30 publishers and 870 subscribers (29 for
+ * each publisher) for a total of 900 peer connections.
+
+ * Due to this, if you have several clients that can only withstand ~100 peer
+ * connections each in order to simulate a 30 participants call you could run
+ * the script with 3 publishers and 29 subscribers per publisher on 10 clients
+ * at the same time.
+ */
+
+// Guest users do not currently work, as they must join the conversation to not
+// be kicked out from the signaling server. However, joining the conversation
+// a second time causes the first guest to be unregistered.
+// Regular users do not need to join the conversation, so the same user can be
+// connected several times to the HPB.
+const user = ''
+const password = ''
+
+const host = 'cloud.example.tld'
```

Comment:
this should be read from the URL ? a cross-domain call is anyway not possible within the browser context

Reply:
Mmm, good idea.